scene/main/viewport.cpp

```diff
@@ -422,19 +385,21 @@ void Viewport::_notification(int p_what) {
 			}
 
 			current_canvas = find_world_2d()->get_canvas();
-			RenderingServer::get_singleton()->viewport_set_scenario(viewport, find_world_3d()->get_scenario());
 			RenderingServer::get_singleton()->viewport_attach_canvas(viewport, current_canvas);
-
-			_update_listener();
 			_update_listener_2d();
+#ifndef _3D_DISABLED
+			RenderingServer::get_singleton()->viewport_set_scenario(viewport, find_world_3d()->get_scenario());
+			_update_listener_3d();
+#endif // _3D_DISABLED
 
 			add_to_group("_viewports");
 			if (get_tree()->is_debugging_collisions_hint()) {
-				//2D
+#ifndef _2D_DISABLED
```

Comment:
There's no `_2D_DISABLED` define, and this isn't done consistently everywhere else, so I wouldn't add it in this PR.